### DISCUSSION TITLE
Image requests from OpenWebUI not reaching Ollama

### DISCUSSION THREAD

**Discussion by loglux** [open-webui/open-webui#8221] [Issues]
I've encountered an issue with OpenWebUI where image requests no longer work, although everything was functioning properly before. This might have started after an OpenWebUI update.

### Details of the setup:

-   **Models Used:** LLaMA3.2-Vision, LLAVA, LLAVA-LLaMA3.
-   **OpenWebUI Environment:** Running in a Docker container on a NAS.
-   **Ollama Environment:** Running locally on a Windows 11 machine.

### Issue Description:

-   Text-based chats work perfectly.
-   When I try to send an image, nothing happens: the spinner shows up, but it gets stuck and no further processing occurs.
-   **Logs:**
    -   No errors in OpenWebUI logs.
    -   Ollama doesn't receive any requests when images are sent.
-   **Tests and Observations:**
    -   Network connectivity between the Docker container and the Windows machine is fine (tested with `curl`).
    -   Models in OpenWebUI are loaded correctly, and text messages are processed as expected.

### What I’ve Tried:

1.  Checked OpenWebUI logs—no relevant errors.
2.  Confirmed that Ollama is running and reachable on the correct port.
3.  Restarted both OpenWebUI and Ollama.
4.  Updated OpenWebUI and Ollama to the latest versions.

It seems like OpenWebUI is not forwarding image requests to Ollama for some reason. Has anyone else encountered this issue or know how to resolve it?

Thanks in advance for any advice or suggestions!

PS.
OpenWebUI Version: v0.5.2 (on OpenWebUI Version: v0.5.1 didn't work too)

**Comment by tjbck** [comment]
More details would be helpful here, I'm unable to reproduce this issue myself:

![image](https://github.com/user-attachments/assets/60cd1947-c73b-4b43-b0a6-d00e842cc69c)

**Reply by loglux** [reply]
![image](https://github.com/user-attachments/assets/881f83f8-64db-475c-89a5-f24e3b8858f3)
It's not sending the screenshot.

**Comment by loglux** [comment]
and there is nothing in the logs

```
INFO  [open_webui.routers.openai] get_all_models()

INFO  [open_webui.routers.ollama] get_all_models()

INFO:     192.168.10.229:0 - "GET /api/models HTTP/1.1" 200 OK

INFO:     192.168.10.229:0 - "GET /api/v1/configs/banners HTTP/1.1" 200 OK

INFO:     192.168.10.229:0 - "GET /api/v1/tools/ HTTP/1.1" 200 OK

INFO:     192.168.10.229:0 - "GET /api/v1/chats/402dd269-ebcd-4c06-b4e2-3ce858c3ac7d HTTP/1.1" 200 OK

INFO:     192.168.10.229:0 - "GET /static/favicon.png HTTP/1.1" 304 Not Modified

INFO:     192.168.10.229:0 - "GET /api/v1/channels/ HTTP/1.1" 200 OK

INFO:     192.168.10.229:0 - "GET /api/v1/chats/all/tags HTTP/1.1" 200 OK

INFO:     192.168.10.229:0 - "GET /api/v1/chats/402dd269-ebcd-4c06-b4e2-3ce858c3ac7d/tags HTTP/1.1" 200 OK

INFO:     192.168.10.229:0 - "GET /api/v1/chats/pinned HTTP/1.1" 200 OK

INFO:     192.168.10.229:0 - "GET /api/v1/users/user/settings HTTP/1.1" 200 OK

INFO:     192.168.10.229:0 - "GET /api/v1/folders/ HTTP/1.1" 200 OK

INFO:     192.168.10.229:0 - "GET /api/v1/chats/?page=1 HTTP/1.1" 200 OK

INFO:     192.168.10.229:0 - "GET /ollama/api/version HTTP/1.1" 200 OK

INFO:     192.168.10.229:0 - "GET /api/version/updates HTTP/1.1" 200 OK

INFO:     192.168.10.229:0 - "GET /api/v1/chats/?page=2 HTTP/1.1" 200 OK
```

**Comment by loglux** [comment]
and now again
the previous attempt, taking a random screenshot from Google worked, the second attempt hangs
![image](https://github.com/user-attachments/assets/eb5659bc-e30a-4781-87a3-df7bd60ffc6e)
And Ollama has received nothing.

**Reply by tjbck** [reply]
Well known model issue here: https://github.com/open-webui/open-webui/discussions/8067

**Comment by loglux** [comment]
It's not just about 2nd image.
![image](https://github.com/user-attachments/assets/4e56d35e-9aa7-4bde-b202-fb3835e08f07)
But this one didn't go, but it was the first!
![image](https://github.com/user-attachments/assets/0f17f889-d3e8-4fff-8fa0-1f87a27eb4f6)

**Reply by loglux** [reply]
And it doesn't depend on a model. 
![image](https://github.com/user-attachments/assets/b8725ea9-6bc0-4e25-a57e-651afca47472)
![image](https://github.com/user-attachments/assets/64142eb4-0020-4486-9c51-c89ddcae78d5)

**Reply by loglux** [reply]
I use a Python script to send 50+ images to ollama/llama3.2-vision with no issue. It can process them consequently without causing an error. But here, OpenWebUI doesn't send anything to Ollama at all.

**Reply by tjbck** [reply]
![image](https://github.com/user-attachments/assets/3a8d2ceb-0a5b-4667-af70-4e79bdeb900a)

Cannot reproduce any of the issues you're facing, definitely continue your troubleshooting journey and keep us updated!

**Comment by loglux** [comment]
Which troubleshooting steps could I do?
The simple log check doesn't show anything.

**Reply by tjbck** [reply]
Most definitely could be related to this: https://github.com/open-webui/open-webui/issues/8074

**Reply by tjbck** [reply]
Also please read and **follow** the issue template including browser/backend logs, so that other people can effectively assist you.

**Comment by loglux** [comment]
![image](https://github.com/user-attachments/assets/b05fa2ce-bec1-4f94-94d8-56ff52bc2559)
and after the request
![image](https://github.com/user-attachments/assets/5e4196e0-f85f-4a8c-b975-0cbecf7a4ca8)

**Comment by loglux** [comment]
![image](https://github.com/user-attachments/assets/4ca07c28-ad37-4bda-8f4b-7218d4072f64)

**Comment by loglux** [comment]
![image](https://github.com/user-attachments/assets/58d151b1-9410-49ba-ae98-471a17dcd31e)

**Comment by loglux** [comment]
**
![image](https://github.com/user-attachments/assets/25b85d16-f4c4-4971-8800-dfea906368b9)
**

**Comment by loglux** [comment]
![image](https://github.com/user-attachments/assets/286a7c8f-3ae6-4d5f-88d0-a8beb8c4166c)

**Comment by loglux** [comment]
```
Actions
 
 
 
      
    await self.simple_response(scope, receive, send, request_headers=headers)

  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/cors.py", line 148, in simple_response

    await self.app(scope, receive, send)

  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 189, in __call__

    with collapse_excgroups():

  File "/usr/local/lib/python3.11/contextlib.py", line 158, in __exit__

    self.gen.throw(typ, value, traceback)

  File "/usr/local/lib/python3.11/site-packages/starlette/_utils.py", line 93, in collapse_excgroups

    raise exc

  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 191, in __call__

    response = await self.dispatch_func(request, call_next)

               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/app/backend/open_webui/main.py", line 723, in inspect_websocket

    return await call_next(request)

           ^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 166, in call_next

    raise RuntimeError("No response returned.")

RuntimeError: No response returned.

ERROR:    Exception in ASGI application

  + Exception Group Traceback (most recent call last):

  |   File "/usr/local/lib/python3.11/site-packages/starlette/_utils.py", line 87, in collapse_excgroups

  |     yield

  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 190, in __call__

  |     async with anyio.create_task_group() as task_group:

  |   File "/usr/local/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 815, in __aexit__

  |     raise BaseExceptionGroup(

  | ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)

  +-+---------------- 1 ----------------

    | Traceback (most recent call last):

    |   File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 401, in run_asgi

    |     result = await app(  # type: ignore[func-returns-value]

    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    |   File "/usr/local/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__

    |     return await self.app(scope, receive, send)

    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    |   File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__

    |     await super().__call__(scope, receive, send)

    |   File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 123, in __call__

    |     await self.middleware_stack(scope, receive, send)

    |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 186, in __call__

    |     raise exc

    |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 164, in __call__

    |     await self.app(scope, receive, _send)

    |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/cors.py", line 93, in __call__

    |     await self.simple_response(scope, receive, send, request_headers=headers)

    |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/cors.py", line 148, in simple_response

    |     await self.app(scope, receive, send)

    |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 189, in __call__

    |     with collapse_excgroups():

    |   File "/usr/local/lib/python3.11/contextlib.py", line 158, in __exit__

    |     self.gen.throw(typ, value, traceback)

    |   File "/usr/local/lib/python3.11/site-packages/starlette/_utils.py", line 93, in collapse_excgroups

    |     raise exc

    |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 191, in __call__

    |     response = await self.dispatch_func(request, call_next)

    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    |   File "/app/backend/open_webui/main.py", line 723, in inspect_websocket

    |     return await call_next(request)

    |            ^^^^^^^^^^^^^^^^^^^^^^^^

    |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 166, in call_next

    |     raise RuntimeError("No response returned.")

    | RuntimeError: No response returned.

    +------------------------------------

During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 401, in run_asgi

    result = await app(  # type: ignore[func-returns-value]

             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__

    return await self.app(scope, receive, send)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__

    await super().__call__(scope, receive, send)

  File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 123, in __call__

    await self.middleware_stack(scope, receive, send)

  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 186, in __call__

    raise exc

  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 164, in __call__

    await self.app(scope, receive, _send)

  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/cors.py", line 93, in __call__

    await self.simple_response(scope, receive, send, request_headers=headers)

  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/cors.py", line 148, in simple_response

    await self.app(scope, receive, send)

  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 189, in __call__

    with collapse_excgroups():

  File "/usr/local/lib/python3.11/contextlib.py", line 158, in __exit__

    self.gen.throw(typ, value, traceback)

  File "/usr/local/lib/python3.11/site-packages/starlette/_utils.py", line 93, in collapse_excgroups

    raise exc

  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 191, in __call__

    response = await self.dispatch_func(request, call_next)

               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/app/backend/open_webui/main.py", line 723, in inspect_websocket

    return await call_next(request)

           ^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 166, in call_next

    raise RuntimeError("No response returned.")

RuntimeError: No response returned.

```

**Comment by loglux** [comment]
## Explanation

The error "QuotaExceededError: Failed to execute 'setItem' on 'Storage': Setting the value of 'chat-input-...' exceeded the quota" means you're trying to store too much data in the browser's local storage for a given key. Browsers have limits on the amount of data you can store in local storage per domain. The typical limit is around 5-10MB, although it can vary.

Here's how to fix this, along with explanations and best practices:

**1. Reduce the Size of Stored Data:**

-   **Minimize the  `input`  object:**  The most likely culprit is the  `input`  object you're storing. Examine its contents. Are there large strings, arrays, or objects that might be unnecessary? Do you  _really_  need to store the entire input history in local storage? Consider storing only essential data. For example, if  `input.prompt`  is the only important part, store only that.

#### css

onChange={(input) => {

if (input.prompt) {

localStorage.setItem(`chat-input-${$chatId}`, JSON.stringify(input.prompt)); // Store only the prompt

} else {

localStorage.removeItem(`chat-input-${$chatId}`);

}

}}

Use code snippets with caution

-   **Compress the data:**  If you absolutely need to store the entire  `input`  object, consider compressing it before storing it in local storage. Libraries like  `lz-string`  can help with this. This will significantly reduce the storage space required.

#### js

import LZString from 'lz-string'; // Install: npm install lz-string

  

onChange={(input) => {

if (input.prompt) {

const compressedData = LZString.compressToUTF16(JSON.stringify(input));

localStorage.setItem(`chat-input-${$chatId}`, compressedData);

} else {

localStorage.removeItem(`chat-input-${$chatId}`);

}

}}

Use code snippets with caution

And when retrieving:

#### js

const compressedData = localStorage.getItem(`chat-input-${$chatId}`);

const input = JSON.parse(LZString.decompressFromUTF16(compressedData));

Use code snippets with caution

**2. Implement a Data Expiration Strategy:**

-   **Limit history:**  Instead of storing the entire chat history, consider storing only the  _last N_  messages. This will keep the local storage size manageable.
-   **Time-based expiration:**  Use a timestamp to track when data was stored. Remove older entries to free up space.

**3. Use a Different Storage Mechanism (if necessary):**

-   **IndexedDB:**  For larger amounts of data, IndexedDB is a more robust solution than localStorage. It's designed for structured data and offers better performance for larger datasets. IndexedDB requires more code but is far more scalable.

**4. Check for Memory Leaks:**

-   Although less likely the direct cause, make sure you aren't inadvertently creating duplicate entries in local storage.

**Example using Time-based Expiration (Combined with Reduced Storage):**

#### css

onChange={(input) => {

if (input.prompt) {

const dataToStore = {

timestamp: Date.now(),

prompt: input.prompt

};

localStorage.setItem(`chat-input-${$chatId}`, JSON.stringify(dataToStore));

  

// Clean up old entries (e.g., keep only entries from the last hour)

for (let i = 0; i < localStorage.length; i++) {

const key = localStorage.key(i);

if (key.startsWith('chat-input-')) {

const storedData = JSON.parse(localStorage.getItem(key));

if (Date.now() - storedData.timestamp > 3600000) { // 1 hour

localStorage.removeItem(key);

}

}

}

} else {

localStorage.removeItem(`chat-input-${$chatId}`);

}

}}

Use code snippets with caution

**Debugging Tips:**

-   **Check localStorage size:**  Open your browser's developer tools (usually by pressing F12), go to the "Application" tab, and look at the "Storage" section. You can see how much space is being used.
-   **Log the size of the  `input`  object:**  Use  `console.log(JSON.stringify(input).length)`  in your  `onChange`  handler to see how large the data is.
-   **Test with smaller data:**  Temporarily reduce the size of the  `input`  object to see if the error disappears.

By combining these techniques, you should be able to resolve the QuotaExceededError and ensure your application functions correctly. Prioritize reducing the amount of data you're storing first, then consider other options if necessary. IndexedDB is highly recommended if you need substantial storage.

Data used to understand this message

Use search instead

AI tools may generate inaccurate info that doesn't represent Google's views. Data sent to Google may be seen by human reviewers to improve this feature.  Open settings  or  learn more

MessageInput.svelte:722 design inspiration

  

Stack table expanded

**Comment by loglux** [comment]
AI Explanation:

### Explanation of the Issue: QuotaExceededError

The error **"QuotaExceededError: Failed to execute 'setItem' on 'Storage': Setting the value of 'chat-input-...' exceeded the quota"** occurs when a program tries to store too much data in the browser's `localStorage`. Browsers impose strict limits on the amount of data that can be stored in `localStorage` per domain, typically around **5-10MB**, though the exact value depends on the browser.

From the information provided above, this issue seems to be caused by the implementation of the program, not the browser. Here's a detailed analysis:

---

### What Happens
1. The program stores user input data (e.g., `chat-input`) in `localStorage` under keys like `chat-input-d7541b61-24bf-4cb7-ae2a-6614e8c56ed2`.
2. Over time, these stored data accumulate and eventually exceed the browser's storage quota.
3. The browser blocks further storage operations and throws the **QuotaExceededError**.

---

### Root Cause
Based on the logs and behavior:
1. **Excessive Data Storage:**
   - The program is attempting to store large objects (e.g., the entire input object) in `localStorage`, which quickly fills up the available space.

2. **No Data Cleanup:**
   - There is no mechanism in the program to remove outdated or unnecessary data from `localStorage`.

3. **Browser Limitations:**
   - The browser's `localStorage` is not designed for large or growing datasets. It enforces a strict size limit to protect user resources.

---

### Why It's a Program Issue, Not a Browser Issue
1. The browser works as intended by enforcing storage limits. These limits are necessary to prevent abuse and protect user resources.
2. The program does not account for these limits:
   - It stores too much data without checking the current storage usage.
   - It lacks mechanisms to manage or optimize storage, such as cleaning up old entries or compressing data.

In summary, this is primarily an issue with the program's design, which needs to handle storage more efficiently.

---

### Proposed Fixes for the Program

#### 1. Minimize Stored Data
- **Store only essential parts of the input object:**
  Instead of saving the entire object, store only the necessary fields, such as the user prompt.
  ```javascript
  onChange = (input) => {
    if (input.prompt) {
      localStorage.setItem(`chat-input-${$chatId}`, JSON.stringify(input.prompt));
    } else {
      localStorage.removeItem(`chat-input-${$chatId}`);
    }
  };
  ```

#### 2. Compress Data
- Use a compression library like `lz-string` to reduce the size of stored data.
  ```javascript
  import LZString from 'lz-string';

  onChange = (input) => {
    if (input.prompt) {
      const compressedData = LZString.compressToUTF16(JSON.stringify(input));
      localStorage.setItem(`chat-input-${$chatId}`, compressedData);
    } else {
      localStorage.removeItem(`chat-input-${$chatId}`);
    }
  };

  // Decompress when retrieving:
  const compressedData = localStorage.getItem(`chat-input-${$chatId}`);
  const input = JSON.parse(LZString.decompressFromUTF16(compressedData));
  ```

#### 3. Implement Data Expiration
- Remove outdated or unnecessary entries from `localStorage` to prevent accumulation.
  ```javascript
  onChange = (input) => {
    if (input.prompt) {
      const dataToStore = {
        timestamp: Date.now(),
        prompt: input.prompt,
      };
      localStorage.setItem(`chat-input-${$chatId}`, JSON.stringify(dataToStore));

      // Clean up old entries (older than 1 hour):
      for (let i = 0; i < localStorage.length; i++) {
        const key = localStorage.key(i);
        if (key.startsWith('chat-input-')) {
          const storedData = JSON.parse(localStorage.getItem(key));
          if (Date.now() - storedData.timestamp > 3600000) { // 1 hour
            localStorage.removeItem(key);
          }
        }
      }
    } else {
      localStorage.removeItem(`chat-input-${$chatId}`);
    }
  };
  ```

#### 4. Use IndexedDB for Larger Data
- If the data is too large for `localStorage`, consider using **IndexedDB**, which is designed for managing larger datasets.

---

### Debugging Tips
1. **Check Storage Usage:**
   - Open Developer Tools (`F12`) > **Application Tab** > **Storage** > **Local Storage**.
   - Inspect how much space is being used and which keys take up the most space.

2. **Log Data Size:**
   - Use `console.log(JSON.stringify(input).length)` to check the size of the data being stored.

3. **Test with Smaller Data:**
   - Temporarily reduce the size of the stored data to verify that this resolves the issue.

---

### Conclusion
The **QuotaExceededError** is not a browser bug but a program design issue. The program must account for browser storage limits by optimizing the size and quantity of stored data. Implementing the proposed fixes will help prevent this error and improve overall application performance.

**Comment by loglux** [comment]
I have updated to 5.3 today. The issue is still here
![image](https://github.com/user-attachments/assets/190cbd68-3c97-4e08-b784-9cf5ef10b30b)
![image](https://github.com/user-attachments/assets/17a45666-91cd-49de-bfbb-8cec2951cd29)

**Comment by ferrouswheel** [comment]
As a new user I ran into this basically immediately, because I tried to upload a local photo that was too large for localStorage.

Why is the image saved in localStorage instead of uploaded? I see that open-webui has an uploads directory...

**Comment by anothercake** [comment]
Hi everyone, 

I wanted to share some insights regarding the 413 error when uploading large image files. I encountered this issue and initially thought it was related to Open WebUI, but after further investigation, I found out that in my case, the problem stemmed from my Nginx proxy server's default request size limits. 

If you're using an Nginx proxy server in front of your Open WebUI server and encounter the 413 error when uploading large files, consider checking and adjusting the **client_max_body_size** setting in your Nginx configuration. Here’s a quick example: 

```
http {
    client_max_body_size 50M; # Adjust this value according to your needs
}
```

Make sure to restart your Nginx server after making changes for them to take effect. 

I hope this helps those who are using an Nginx proxy in front of their Open WebUI setup! 

Cheers!

**Reply by hidara2000** [reply]
You legend! this worked for me

**Reply by srajangarg** [reply]
damn, thanks! i assumed it was an openwebui issue

**Reply by sybadm** [reply]
Even I thought it was openwebui

**Comment by loglux** [comment]
**Clarification: Why This Is a Two-Problem Issue**

The image upload problem in OpenWebUI stems from **two separate issues**:

1. **Nginx Request Size Limitation:**

Nginx has a default limit on the size of HTTP requests, which causes the **413 Request Entity Too Large** error for large uploads.

However, this limitation appears to affect OpenWebUI specifically, as other services coordinated through the same Nginx setup function without requiring adjustments to `client_max_body_size`.

This can be fixed by increasing the `client_max_body_size` parameter in the Nginx configuration (e.g., to 50MB).

2. **OpenWebUI's Inefficient Data Handling:**

Even without Nginx, directly connecting to OpenWebUI still causes image upload issues due to the way it handles data.

Browser logs show **QuotaExceededError** because too much data is stored in `localStorage`.

The program needs better storage management, such as limiting data size, compressing data, or using IndexedDB for large datasets.

**Conclusion:** Fixing Nginx allows large requests to pass, but OpenWebUI also requires internal refactoring to handle uploads efficiently. Both issues must be addressed for a complete solution.
